addons/purchase/models/purchase.py

```diff
@@ -765,7 +765,7 @@ def _create_update_date_activity(self, updated_dates):
         activity = self.activity_schedule(
             'mail.mail_activity_data_warning',
             summary=_("Date Updated"),
-            user_id=self.user_id.id
+            user_id=self.user_id.id or SUPERUSER_ID
```

Comment:
I would pass False instead to fall back to env.user in activity_schedule()

Reply:
@Whenrow Passing False or self.user_id.id when there is no user_id is the same. But thanks to your remark I noticed that the real issue was coming function ```activity_schedule```